deployment/local-compose/README.md

```diff
@@ -0,0 +1,159 @@
+# Run a local OpenWPM crawl using Docker Compose
+
+Documentation and scripts to launch an OpenWPM crawl using Docker Compose locally.
+
+- [Run a local OpenWPM crawl using Docker Compose](#run-a-local-openwpm-crawl-using-docker-compose)
+  - [Prerequisites](#prerequisites)
+    - [Build Docker image](#build-docker-image)
+    - [Set up a mock S3 service](#set-up-a-mock-s3-service)
+    - [Deploy the redis server which we use for the work queue](#deploy-the-redis-server-which-we-use-for-the-work-queue)
+    - [Adding sites to be crawled to the queue](#adding-sites-to-be-crawled-to-the-queue)
+  - [Starting the crawl](#starting-the-crawl)
+  - [Monitor Job](#monitor-job)
+    - [Queue status](#queue-status)
+    - [Job status](#job-status)
+    - [View Job logs](#view-job-logs)
+  - [Inspecting crawl results](#inspecting-crawl-results)
+  - [Clean up all created containers](#clean-up-all-created-containers)
+## Prerequisites
+
+Install Docker and Docker Compose. Note that
+Docker for Mac & Windows include Docker Compose
+
+
+For the remainder of these instructions, you are assumed to be in the `deployment/local-compose/` folder.
+
+### Build Docker image
+
+Make sure that you have an up to date docker image locally:
+
+```
+docker pull openwpm/openwpm
+docker tag openwpm/openwpm openwpm
+```
+
+Alternatively, you can build the image from a local OpenWPM code repository:
+
+```
+cd path/to/OpenWPM
+docker build -t openwpm .
+cd -
+```
+### Set up a mock S3 service
+
+```
+docker-compose up -d localstack
+```
+
+### Deploy the redis server which we use for the work queue
+
+```
+docker-compose up -d redis
+```
+
+### Adding sites to be crawled to the queue
+
+First set the REDIS_CONTAINER enviroment variable:
+```
+export REDIS_CONTAINER=$(docker-compose ps -q redis)
+```
+Create a comma-separated site list as per:
+
+```
+echo "1,http://www.example.com
+2,http://www.example.org
+3,http://www.princeton.edu
+4,http://citp.princeton.edu/" > site_list.csv
+
+../load_site_list_into_redis.sh crawl-queue site_list.csv 
+```
+This command will fail but leave you with a `joblist.txt` in your working directory.
```

Comment:
Please refactor this file to work without throwing an error. Perhaps you can check for the existence of `REDIS_CONTAINER` and skip the kubernetes commands in favor of the docker commands below.

Reply:
I've adapted the script and tested it for the docker-compose case, however I could test if it still works for kubectl.
I'm assuming it does since the logic should be unchanged in that case.